modules/apps/calendar/calendar-api/src/main/java/com/liferay/calendar/util/CalendarBookingIterator.java

```diff
@@ -147,13 +151,23 @@ private boolean _isExceededCount() {
 		return false;
 	}
 
-	private DateValue _toDateValue(long time) {
-		Calendar jCalendar = JCalendarUtil.getJCalendar(
-			time, _getTimeZone(_calendarBooking));
+	private DateValue _toDateValue(long time) throws ParseException {
+		SimpleDateFormat simpleDateFormat = new SimpleDateFormat(
+			DateUtil.ISO_8601_PATTERN);
+
+		simpleDateFormat.setTimeZone(_displayTimeZone);
+
+		String dateString = simpleDateFormat.format(new Date(time));
+
+		Date date = DateUtil.parseDate(
+			DateUtil.ISO_8601_PATTERN, dateString, LocaleUtil.getDefault());
+
+		Calendar calendar = JCalendarUtil.getJCalendar(date.getTime());
 
-		return new DateValueImpl(
-			jCalendar.get(Calendar.YEAR), jCalendar.get(Calendar.MONTH) + 1,
-			jCalendar.get(Calendar.DAY_OF_MONTH));
+		return new DateTimeValueImpl(
+			calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH),
```

Comment:
Why this `calendar.get(Calendar.MONTH) + 1` was replaced by this `calendar.get(Calendar.MONTH)`?

Wouldn't we need to increment as well?